subsys/bluetooth/shell/bt.c

```diff
@@ -1245,6 +1247,289 @@ static int cmd_adv_info(const struct shell *shell, size_t argc, char *argv[])
+	if (err) {
+		shell_error(shell,
+			    "Failed to set periodic advertising data (%d)",
+			    err);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+#endif /* CONFIG_BT_PER_ADV */
+
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+static struct bt_le_per_adv_sync *per_adv_syncs[CONFIG_BT_PER_ADV_SYNC_MAX];
+static struct bt_le_per_adv_sync *selected_per_adv_sync;
+
+static void per_adv_sync_sync_cb(struct bt_le_per_adv_sync *sync,
+				 struct bt_le_per_adv_sync_synced_info *info)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	if (selected_per_adv_sync != sync) {
+		shell_error(ctx_shell, "Invalid per adv sync object");
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	shell_print(ctx_shell, "PER_ADV_SYNC[%u]: [DEVICE]: %s synced, "
+		    "Interval 0x%04x (%u ms), PHY %s",
+		    bt_le_per_adv_sync_get_index(sync), le_addr,
+		    info->interval, info->interval * 5 / 4, phy2str(info->phy));
+}
+
+static void per_adv_sync_terminated_cb(
+	struct bt_le_per_adv_sync *sync,
+	const struct bt_le_per_adv_sync_term_info *info)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	if (selected_per_adv_sync != sync) {
+		shell_error(ctx_shell, "Invalid per adv sync object");
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	shell_print(ctx_shell, "PER_ADV_SYNC[%u]: [DEVICE]: %s sync terminated",
+		    bt_le_per_adv_sync_get_index(sync), le_addr);
+	selected_per_adv_sync = NULL;
+}
+
+static void per_adv_sync_recv_cb(
+	struct bt_le_per_adv_sync *sync,
+	const struct bt_le_per_adv_sync_recv_info *info,
+	struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	if (selected_per_adv_sync != sync) {
+		shell_error(ctx_shell, "Invalid per adv sync object");
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	shell_print(ctx_shell, "PER_ADV_SYNC[%u]: [DEVICE]: %s, tx_power %i, "
+		    "RSSI %i, CTE %u, data length %u",
+		    bt_le_per_adv_sync_get_index(sync), le_addr, info->tx_power,
+		    info->rssi, info->cte_type, buf->len);
+}
+
+static struct bt_le_per_adv_sync_cb per_adv_sync_cb = {
+	.synced = per_adv_sync_sync_cb,
+	.term = per_adv_sync_terminated_cb,
+	.recv = per_adv_sync_recv_cb
+};
+
+static int cmd_per_adv_sync_create(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	int err;
+	struct bt_le_per_adv_sync_param create_params = { 0 };
+	uint32_t options = 0;
+	struct bt_le_per_adv_sync *free_per_adv_sync = NULL;
+
+	for (int i = 0; i < ARRAY_SIZE(per_adv_syncs); i++) {
+		if (per_adv_syncs[i] == NULL) {
+			free_per_adv_sync = per_adv_syncs[i];
+			break;
+		}
+	}
+
+	if (!free_per_adv_sync) {
+		shell_error(shell, "Cannot create more per adv syncs");
+		return -ENOEXEC;
+	}
+
+	err = bt_addr_le_from_str(argv[1], argv[2], &create_params.addr);
+	if (err) {
+		shell_error(shell, "Invalid peer address (err %d)", err);
+		return -ENOEXEC;
+	}
+
+	create_params.timeout = 1000; /* 10 seconds */
+	create_params.skip = 10;
+
+	if (argc > 3) {
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/blob/1b856d112e60285c0e66ceb93ab770ad9429cb94/subsys/bluetooth/shell/bt.c#L735

Possible way to make these options independent on order, and optional if the only argument you wish to provide is 'aod_1us' for example.

Reply:
Good idea, added for timeout and skip :)